composite_handler/api/views.py

```diff
@@ -0,0 +1,62 @@
+from api.models import Submission
+from api.serializers import SubmissionCollectionSerializer, SubmissionMemberSerializer
+from django_q.tasks import async_task
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+
+class SubmissionAPIQueryView(APIView):
+    def get(self, request, *args, **kwargs):
+        try:
+            submission = Submission.objects.get(id=kwargs["id"])
+            if submission.status == "PENDING":
+                return Response(status=status.HTTP_200_OK)
+            elif submission.status == "COMPLETED":
+                return Response(status=status.HTTP_303_SEE_OTHER, headers={"Location": submission.dlcs_uri})
+            elif submission.status == "ERROR":
+                return Response({"Error": submission.error}, status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+        except Submission.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+
+class SubmissionAPICollectionView(APIView):
+    def post(self, request, *args, **kwargs):
+        data = {
+            "payload": request.data,
+            "customer": kwargs["customer"],
+        }
+        serializer = SubmissionCollectionSerializer(data=data)
+        return _build_submissions_creation_response(request, serializer)
+
+
+class SubmissionAPIMemberView(APIView):
+    def post(self, request, *args, **kwargs):
+        data = {
+            "payload": request.data,
+            "customer": kwargs["customer"],
+            "space": kwargs["space"],
+            "image": kwargs["image"]
+        }
+        serializer = SubmissionMemberSerializer(data=data)
+        return _build_submissions_creation_response(request, serializer)
+
+
+def _build_submissions_creation_response(request, serializer):
```

Comment:
This is (to an extent) duplicating functionality you'd get from a generic API view like `CreateAPIView`, which would deal with the serializer validation/save, and has a `get_success_headers` method which could be overwritten if the `Location` it provides is incorrect. You'd have to create a custom subclass of the view to return the right status code, which I think is the only gotcha about this approach. The creation of the async task could be moved to a post-save hook off the submission model, which may be more appropriate anyway given that I assume that the task should run after the creation of any submission. 

I don't have any objection to the way this has been done, I'd just refactor if I was expecting to write more of these endpoints, and I could give you a hand with that if it's something you were going to do.